### DISCUSSION TITLE
Unable to install ecole on a windows machine

### DISCUSSION THREAD

**Discussion by rs-DOS** [ds4dm/ecole#260] [Questions]
I am unable to install ecole on a windows machine. cmake's output files are unavailable for review(I can't find them in the folder path given by the error message).
The complete error message has been pasted below:

(base) C:\Users\HP>pip install ecole
WARNING: Ignoring invalid distribution -umpy (c:\users\hp\anaconda3\lib\site-packages)
WARNING: Ignoring invalid distribution -tatsmodels (c:\users\hp\anaconda3\lib\site-packages)
WARNING: Ignoring invalid distribution -rotobuf (c:\users\hp\anaconda3\lib\site-packages)
WARNING: Ignoring invalid distribution -umpy (c:\users\hp\anaconda3\lib\site-packages)
WARNING: Ignoring invalid distribution -tatsmodels (c:\users\hp\anaconda3\lib\site-packages)
WARNING: Ignoring invalid distribution -rotobuf (c:\users\hp\anaconda3\lib\site-packages)
Collecting ecole
  Using cached ecole-0.7.3.tar.gz (366 kB)
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
    Preparing wheel metadata ... done
Requirement already satisfied: numpy>=1.4 in c:\users\hp\anaconda3\lib\site-packages (from ecole) (1.21.2)
Building wheels for collected packages: ecole
  Building wheel for ecole (PEP 517) ... error
  ERROR: Command errored out with exit status 1:
   command: 'c:\users\hp\anaconda3\python.exe' 'c:\users\hp\anaconda3\lib\site-packages\pip\_vendor\pep517\in_process\_in_process.py' build_wheel 'C:\Users\HP\AppData\Local\Temp\tmpds98_ev1'
       cwd: C:\Users\HP\AppData\Local\Temp\pip-install-_v5uwdi7\ecole_66c5b43deb5443afa2e75e15c777a1e2
  Complete output (94 lines):
  Not searching for unused variables given on the command line.
  -- The CXX compiler identification is MSVC 19.29.30133.0
  -- Detecting CXX compiler ABI info
  -- Detecting CXX compiler ABI info - done
  -- Check for working CXX compiler: C:/Program Files (x86)/Microsoft Visual Studio/2019/Community/VC/Tools/MSVC/14.29.30133/bin/Hostx86/x64/cl.exe - skipped
  -- Detecting CXX compile features
  -- Detecting CXX compile features - done
  -- Configuring done
  -- Generating done
  -- Build files have been written to: C:/Users/HP/AppData/Local/Temp/pip-install-_v5uwdi7/ecole_66c5b43deb5443afa2e75e15c777a1e2/_cmake_test_compile/build
  -- Ecole version 0.7.3
  -- The CXX compiler identification is MSVC 19.29.30133.0
  -- Detecting CXX compiler ABI info
  -- Detecting CXX compiler ABI info - done
  -- Check for working CXX compiler: C:/Program Files (x86)/Microsoft Visual Studio/2019/Community/VC/Tools/MSVC/14.29.30133/bin/Hostx86/x64/cl.exe - skipped
  -- Detecting CXX compile features
  -- Detecting CXX compile features - done
  -- IPO enabled
  -- Looking for C++ include pthread.h
  -- Looking for C++ include pthread.h - not found
  -- Found Threads: TRUE
  -- Downloading xtl
  -- Building xtl
  -- Downloading xsimd
  -- Building xsimd
  -- Downloading xtensor
  -- Building xtensor
  -- Performing Test arch_native_supported
  -- Performing Test arch_native_supported - Failed
  -- Downloading span-lite
  -- Building span-lite
  -- Downloading range-v3
  -- Building range-v3
  -- Downloading fmt
  -- Building fmt
  -- Downloading robin_hood
  -- Building robin_hood
  -- Found Python: c:/users/hp/anaconda3/python.exe (found version "3.8.3") found components: Interpreter Development NumPy Development.Module Development.Embed
  -- Downloading pybind11
  -- Building pybind11
  -- Performing Test HAS_MSVC_GL_LTCG
  -- Performing Test HAS_MSVC_GL_LTCG - Success
  -- Downloading xtensor-python
  -- Building xtensor-python
  -- Configuring done
  -- Generating done
  -- Build files have been written to: C:/Users/HP/AppData/Local/Temp/pip-install-_v5uwdi7/ecole_66c5b43deb5443afa2e75e15c777a1e2/_skbuild/win-amd64-3.8/cmake-build
  [1/46] cmd.exe /C "cd /D C:\Users\HP\AppData\Local\Temp\pip-install-_v5uwdi7\ecole_66c5b43deb5443afa2e75e15c777a1e2\_skbuild\win-amd64-3.8\cmake-build\libecole && C:\Users\HP\AppData\Local\Temp\pip-build-env-d7jz0q9c\overlay\Lib\site-packages\cmake\data\bin\cmake.exe -D SOURCE_FILE="C:/Users/HP/AppData/Local/Temp/pip-install-_v5uwdi7/ecole_66c5b43deb5443afa2e75e15c777a1e2/libecole/include/ecole/version.hpp.in" -D TARGET_FILE="C:/Users/HP/AppData/Local/Temp/pip-install-_v5uwdi7/ecole_66c5b43deb5443afa2e75e15c777a1e2/_skbuild/win-amd64-3.8/cmake-build/libecole/include/ecole/version.hpp" -D Ecole_VERSION_MAJOR="0" -D Ecole_VERSION_MINOR="7" -D Ecole_VERSION_PATCH="3" -D Ecole_VERSION_REVISION="" -D Ecole_BUILD_TYPE="Release" -D Ecole_BUILD_OS="Windows-10.0.19042" -D Ecole_BUILD_TIME="2021-10-04T20:38:17" -D Ecole_BUILD_COMPILER="MSVC-19.29.30133.0" -P C:/Users/HP/AppData/Local/Temp/pip-install-_v5uwdi7/ecole_66c5b43deb5443afa2e75e15c777a1e2/cmake/CreateVersionFile.cmake > /dev/null"
  FAILED: libecole/CMakeFiles/ecole-lib-version C:/Users/HP/AppData/Local/Temp/pip-install-_v5uwdi7/ecole_66c5b43deb5443afa2e75e15c777a1e2/_skbuild/win-amd64-3.8/cmake-build/libecole/CMakeFiles/ecole-lib-version
  cmd.exe /C "cd /D C:\Users\HP\AppData\Local\Temp\pip-install-_v5uwdi7\ecole_66c5b43deb5443afa2e75e15c777a1e2\_skbuild\win-amd64-3.8\cmake-build\libecole && C:\Users\HP\AppData\Local\Temp\pip-build-env-d7jz0q9c\overlay\Lib\site-packages\cmake\data\bin\cmake.exe -D SOURCE_FILE="C:/Users/HP/AppData/Local/Temp/pip-install-_v5uwdi7/ecole_66c5b43deb5443afa2e75e15c777a1e2/libecole/include/ecole/version.hpp.in" -D TARGET_FILE="C:/Users/HP/AppData/Local/Temp/pip-install-_v5uwdi7/ecole_66c5b43deb5443afa2e75e15c777a1e2/_skbuild/win-amd64-3.8/cmake-build/libecole/include/ecole/version.hpp" -D Ecole_VERSION_MAJOR="0" -D Ecole_VERSION_MINOR="7" -D Ecole_VERSION_PATCH="3" -D Ecole_VERSION_REVISION="" -D Ecole_BUILD_TYPE="Release" -D Ecole_BUILD_OS="Windows-10.0.19042" -D Ecole_BUILD_TIME="2021-10-04T20:38:17" -D Ecole_BUILD_COMPILER="MSVC-19.29.30133.0" -P C:/Users/HP/AppData/Local/Temp/pip-install-_v5uwdi7/ecole_66c5b43deb5443afa2e75e15c777a1e2/cmake/CreateVersionFile.cmake > /dev/null"
  The system cannot find the path specified.
  [2/46] Copying Python files
  ninja: build stopped: subcommand failed.
    File "C:\Users\HP\AppData\Local\Temp\pip-build-env-d7jz0q9c\overlay\Lib\site-packages\skbuild\setuptools_wrap.py", line 593, in setup
      cmkr.make(make_args, env=env)
    File "C:\Users\HP\AppData\Local\Temp\pip-build-env-d7jz0q9c\overlay\Lib\site-packages\skbuild\cmaker.py", line 504, in make
      raise SKBuildError(

  --------------------------------------------------------------------------------
  -- Trying "Ninja (Visual Studio 16 2019 x64 v142)" generator
  --------------------------------
  ---------------------------
  ----------------------
  -----------------
  ------------
  -------
  --
  --
  -------
  ------------
  -----------------
  ----------------------
  ---------------------------
  --------------------------------
  -- Trying "Ninja (Visual Studio 16 2019 x64 v142)" generator - success
  --------------------------------------------------------------------------------

  Configuring Project
    Working directory:
      C:\Users\HP\AppData\Local\Temp\pip-install-_v5uwdi7\ecole_66c5b43deb5443afa2e75e15c777a1e2\_skbuild\win-amd64-3.8\cmake-build
    Command:
      cmake 'C:\Users\HP\AppData\Local\Temp\pip-install-_v5uwdi7\ecole_66c5b43deb5443afa2e75e15c777a1e2' -G Ninja '-DCMAKE_INSTALL_PREFIX:PATH=C:\Users\HP\AppData\Local\Temp\pip-install-_v5uwdi7\ecole_66c5b43deb5443afa2e75e15c777a1e2\_skbuild\win-amd64-3.8\cmake-install\python\src' '-DPYTHON_EXECUTABLE:FILEPATH=c:\users\hp\anaconda3\python.exe' -DPYTHON_VERSION_STRING:STRING=3.8.3 '-DPYTHON_INCLUDE_DIR:PATH=c:\users\hp\anaconda3\Include' '-DPYTHON_LIBRARY:FILEPATH=c:\users\hp\anaconda3\libs\python38.lib' -DSKBUILD:INTERNAL=TRUE '-DCMAKE_MODULE_PATH:PATH=C:\Users\HP\AppData\Local\Temp\pip-build-env-d7jz0q9c\overlay\Lib\site-packages\skbuild\resources\cmake' -DCMAKE_BUILD_TYPE:STRING=Release

  Traceback (most recent call last):

  An error occurred while building with CMake.
    Command:
      cmake --build . --target install --config Release --
    Source directory:
      C:\Users\HP\AppData\Local\Temp\pip-install-_v5uwdi7\ecole_66c5b43deb5443afa2e75e15c777a1e2
    Working directory:
      C:\Users\HP\AppData\Local\Temp\pip-install-_v5uwdi7\ecole_66c5b43deb5443afa2e75e15c777a1e2\_skbuild\win-amd64-3.8\cmake-build
  Please see CMake's output for more information.
  ----------------------------------------
  ERROR: Failed building wheel for ecole
Failed to build ecole
ERROR: Could not build wheels for ecole which use PEP 517 and cannot be installed directly
WARNING: Ignoring invalid distribution -umpy (c:\users\hp\anaconda3\lib\site-packages)
WARNING: Ignoring invalid distribution -tatsmodels (c:\users\hp\anaconda3\lib\site-packages)
WARNING: Ignoring invalid distribution -rotobuf (c:\users\hp\anaconda3\lib\site-packages)
WARNING: Ignoring invalid distribution -umpy (c:\users\hp\anaconda3\lib\site-packages)
WARNING: Ignoring invalid distribution -tatsmodels (c:\users\hp\anaconda3\lib\site-packages)
WARNING: Ignoring invalid distribution -rotobuf (c:\users\hp\anaconda3\lib\site-packages)

**Comment by AntoinePrv** [comment]
We currently don't support Windows. Any help is appreciated.
